kernel/syscall.c

```diff
@@ -3169,6 +3169,20 @@ long myst_syscall_mbind(
     return ret;
 }
 
+long myst_syscall_get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    long params[6] = {
```

Comment:
From Linux manual:

If nodemask is not NULL, then the nodemask associated with the policy will be
stored in the location pointed to by this argument. maxnode specifies the number
of node IDs that can be stored into nodemask—that is, the maximum node ID plus
one. The value specified by maxnode is always rounded to a multiple of
sizeof(unsigned long)*8.

The app might provide a maxnode value that's not a multiple of `sizeof(unsigned long)*8`. Round it up according to the Linux manual before calling into the target will simplify OCALL implementation.

Reply:
Thank you for your suggestion, I have rounded up the value here

Reply:
Thinking out load --- given we're rounding up the size, I'm wondering would there be a case of `nodemask` being overflowed. This may also apply to the behavior of the syscall itself --- how does the syscall behave if the user is passing unrounded `maxnode`?  Maybe this have been guarded by the memory alignment? Can we confirm this behavior?

Reply:
Thank you for helping me with this case ming-wei, have added our added testcases here